README.md

```diff
@@ -1,13 +1,408 @@
+```
+
+#### ::each()
+
+#### ::filter()
+
+#### ::slice()
+
+#### ::map()
+
+#### ::tmap()
+
+#### ::reduceLeft()
+
+#### ::reduceRight()
+
+#### ::treduce()
+
+#### ::foldLeft()
+
+#### ::scanLeft()
+
+#### ::scanRight
+
+### Terminal operators
+
+#### ::vector()
+
+**Method**:
+```cpp
+std::vector<T> vector();
+```
+
+Returns the Collection as a vector object. Since Collections are stored as vectors, this method simply returns the internal data object.
+
+*Example*:
+```cpp
+std::vector<int> v = Collection<int>(5).vector(); //returns a vector of 5 zeros
+```
+
+#### ::list()
+
+**Method**:
+```cpp
+std::list<T> list();
+```
+
+Returns the Collection as a list object. Since Collections are stored as vectors, this method simply returns the internal data object which is then converted to a list by using <section>std::begin(Data)</section> and <section>std::end(Data)</section> to a <section>std::list</section>.
+
+*Example*:
+```cpp
+std::list<int> l = Collection<int>(5).list(); //returns a list of 5 zeros
+```
+
+#### ::size()
+
+**Method**:
+```cpp
+int size();
+```
+
+Returns an <section>int</section> with the current size of the collection. This is executed by running the <section>size()</section> method on the internal vector which holds the internal data.
+
+*Example*:
+```cpp
+auto col = Collection<int>(5);
+std::cout<< collec.size() << std::endl;
+```
+
+#### ::print()
+
+**Method**:
+```cpp
+void print();
+```
+
+This method is also a terminal operator but does not return anything, but instead prints the Collection to the `std::cout`.
+
+*Example*:
+```cpp
+auto col = Collection<int>(5);
+col.print();
+```
+
+*Output*:
+```
+[0,0,0,0,0]
+```
+
+#### ::head()
+
+**Method**:
+```cpp
+T head();
+```
+
+This method is a terminal operator which returns the first value of the Collection. It returns based on the type of the values stored on the Collection.
+
+*Example*:
+```cpp
+auto col = range(1,101);
+std::cout<< col.head(); << "\n";
+```
+
+*Output*:
+```
+1
+```
+
+#### ::last()
+
+**Method**:
+```cpp
+T last();
+```
+
+This method is a terminal operator which returns the last value of the Collection. It returns based on the type of the values stored on the Collection.
+
+*Example*:
+```cpp
+auto col = range(1,101);
+std::cout<< col.last(); << "\n";
+```
+
+*Output*:
+```
+100
+```
+
+
+## Streams
+
+The second object in the C++ Collections library that we would like to expand on is the Stream object. Streams are developed by having a head of anytype, a tail which is initially set to a nullptr and a generator function. The initial stream allows users to run `take()`, `head()`, `tail()`, `filter()`, and `map()` on it. Yet Streams also allow the development of infinite lists using the `from()` as well. As the Stream object is developed, the documentation will be updated correspondingly. 
+
+## Development Suport
+
+If you are interested in contributing, 
+```
+g++ -std=c++11 -Wall tests/<FILE>.cpp
+```
+
+## Tests
```

Comment:
Okay I'll put that all under something separate - that or we could do a separate document? I think just a section will work though.